### ISSUE TITLE
Loop variable, in alias, incorrectly flagged as unassigned

### ISSUE THREAD

**Issue by pbx** [koalaman/shellcheck#1990]
#### Bug
- Rule ID: SC2154
- My shellcheck version: 0.7.1
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Snippet that triggers the problem:
```sh
#!/bin/bash
alias repro='for x in a b c; do echo $x; done;'
```

#### Here's what shellcheck currently says:
```
shellcheck: warning
warning - x is referenced but not assigned. [SC2154]
```
#### Here's what I wanted or expected to see:
(No warning)

**Comment by forthrin** [comment]
Also happens for `g` here when inside alias:

`alias foo='while read -r f; do g=foo; echo "$f" "$g"; done'`
